activemq-broker/src/main/java/org/apache/activemq/broker/region/Queue.java

```diff
@@ -2387,8 +2495,8 @@ private QueueMessageReference getMatchingMessage(MessageDispatchNotification mes
                     messages.reset();
                     while (messages.hasNext()) {
                         MessageReference node = messages.next();
-                        messages.remove();
```

Comment:
Why move this?

Reply:
The idea of this change is to delete only the message that we are looking for. Because deleting all messages up to the one we need prevents reading them in future. Basically if we delete them here, we will not be able to get them anymore.

Reply:
And? That's now unneeded change in behaviour

Reply:
I am not sure why.  this method is used to "deliver" a specific message by id to a specific consumer. we still want to have the access to all other messages. in case we want to "deliver" them too, or in case there is a real consumer